CodePush.m

```diff
@@ -468,6 +529,55 @@ - (void)savePendingUpdate:(NSString *)packageHash
 }
 
 /*
+ * This method is checks if a new status update exists (new version was installed, 
+ * or an update failed) and return its details (version label, status).
+ */
+RCT_EXPORT_METHOD(getNewStatusReport:(RCTPromiseResolveBlock)resolve
+                            rejecter:(RCTPromiseRejectBlock)reject)
+{
+    // Check if the current appVersion has been reported.
+    NSString *appVersion = [[CodePushConfig current] appVersion];
+    if ([self isDeploymentStatusNotYetReported:appVersion]) {
+        [self recordDeploymentStatusReported:appVersion
+                                      status:DeploymentSucceeded];
+        resolve(@{ @"appVersion": appVersion });
+        return;
+    }
+    
+    // Check if there was a rollback that was not yet reported
+    NSUserDefaults *preferences = [NSUserDefaults standardUserDefaults];
+    NSMutableArray *failedUpdates = [preferences objectForKey:FailedUpdatesKey];
```

Comment:
The scenario where there isn't anything to report is exactly what I'd like to optimize. Ideally, calling notifyApplicationReady should have an extremely low impact on an app that hasn't even released an update yet. Maybe what you have is fine, i just wanted to mention the desire to keep app start as lean as possible, since this will run every time, but be unnecessary most of the time.